common/src/main/java/tachyon/retry/CountingRetry.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.retry;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * An option which allows retrying based on maximum count.
+ */
+public class CountingRetry implements RetryPolicy {
+
+  private final int mMaxRetries;
+  private int mCount = -1;
```

Comment:
That would just return an invalid retryCount in getRetryCount. 
By saying make it consistent
if mCount = 0 then
public int getRetryCount() { return mCount == 0 ? 0 : mCount - 1; } 
since incr would have happened.

I have updated the code and the test